File: Models/CogbytesSearch.cs
using Microsoft.AspNetCore.Identity;

namespace AngularSPAWebAPI.Models
{
    public class CogbytesSearch
    {
        public int?[] RepID { get; set; }
        public string Keywords { get; set; }
        public string DOI { get; set; }
        public int?[] AuthorID { get; set; }
        public int?[] PIID { get; set; }

        public int?[] TaskID { get; set; }
        public int?[] SpecieID { get; set; }
        public int?[] SexID { get; set; }
        public int?[] StrainID { get; set; }
        public int?[] GenoID { get; set; }
        public int?[] AgeID { get; set; }

        public int? YearFrom { get; set; }
        public int? YearTo { get; set; }
        public string Intervention { get; set; }

        public int?[] FileTypeID { get; set; }
        public int?[] DiseaseID { get; set; }
        public int?[] SubModelID { get; set; }
        public int?[] RegionID { get; set; }
        public int?[] SubRegionID { get; set; }
        public int?[] CellTypeID { get; set; }
        public int?[] MethodID { get; set; }
        public int?[] SubMethodID { get; set; }
        public int?[] TransmitterID { get; set; }
    }
}

File: src/app/models/cogbytesSearch.ts
export interface CogbytesSearch {

     repID: Array<number>;
     keywords: string;
     doi: string;
     authorID: Array<number>;
     psID: Array<number>;

     taskID: Array<number>;
     specieID: Array<number>;
     sexID: Array<number>;
     strainID: Array<number>;
     genoID: Array<number>;
     startAge: number | null;
     endAge: number | null;

     yearFrom: any;
     yearTo: any

     intervention: string;

     fileTypeID: Array<number>;

     diseaseID: Array<number>;
     subModelID: Array<number>;
     regionID: Array<number>;
     subRegionID: Array<number>;
     cellTypeID: Array<number>;
     methodID: Array<number>;
     subMethodID: Array<number>;
     transmitterID: Array<number>;
}
